### PR TITLE
DG Bug Fixes and Enhancements

### LINKED ISSUE #294: [PE-D][Tester D] phone validation may be too strict
might want to allow something more of a free-flow text box as suggested by 2103 website

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: weiseng18/ped#9

### PR DESCRIPTION
**Pull request by BuffWuff1712**
fixes #294

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -965,7 +946,29 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
-## **Appendix: Instructions for manual testing**
+## **Appendix B: Planned Enhancements**
+
+As we continue to develop PROperty, we are focused on refining and expanding its capabilities to better meet user needs. 
+Through identifying current limitations, we’re able to target areas that can benefit most from enhancement. 
+This section highlights our planned improvements, showcasing the steps we're taking to deliver a more comprehensive and effective solution for property agents.
+
+
+### Enhance Phone Numbers to Accept Special Characters
```
Comment (Vatinius) [inline docs/DeveloperGuide.md:956]: Maybe we could enumerate the title like the example they gave us (see screenshot above).

**Review by Vatinius — COMMENTED** [review]
Made some comments!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (BuffWuff1712) [reply docs/DeveloperGuide.md:956]: okies will do

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -158,95 +186,48 @@ Classes used by multiple components are in the `seedu.address.commons` package.
...
 
-#### Design considerations:
+`ListingCommandsParser` first obtains the user input string and ensures that the suffix of the command matches 'add' or 'delete'. 
+ If the suffix does not match either, `ListingCommandsParser` throws a ParseException. Otherwise,
+ `ListingCommandsParser` creates `ListingAddCommandParser` to parse the arguments.
 
-**Aspect: How undo & redo executes:**
+`ListingAddCommandParser` then extracts the index and the values corresponding to the prefixes `t/` and `a/`.
+`ListingAddCommandParser` ensures that:
+- Provided index is within a valid range, meaning it cannot be less than 1 or greater than the number of clients in the currently displayed list.
```
[the remark this answers is not in the document]
Comment (BuffWuff1712) [reply docs/DeveloperGuide.md:220]: I think we will stick to client to be consistent with the UG too

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -965,7 +946,29 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 --------------------------------------------------------------------------------------------------------------------
 
-## **Appendix: Instructions for manual testing**
+## **Appendix B: Planned Enhancements**
+
```
[the remark this answers is not in the document]
Comment (Tkaixiang) [reply docs/DeveloperGuide.md:950]: fixed by @BuffWuff1712

**Review by Tkaixiang — APPROVED** [review]
lgtm

**Pull request merged**